### ISSUE TITLE
CSE 3.0.2 cluster upgrade fails when there are OLM pods on the cluster

### ISSUE THREAD

**Issue by dmbuil** [vmware/container-service-extension#1102]
I am using CSE 3.0.2 with VCD 10.2.2 and trying to upgrade TKG+ 1.18 templates to TKG+ 1.19 (both rev.3).
These clusters have OLM deployed (OLM v0.18.1 / v0.18.2).

When I perform an `vcd cse cluster upgrade ...`, the process fails with the following error (excerpt from `cse-server-debug.log`):

```
...
21-07-07 09:58:44 | cluster_service:2277 - _execute_script_in_nodes | Request Id: fc8e70a8-38ae-4a2f-8649-3bf2edf0d5b0 | DEBUG :: 1
21-07-07 09:58:44 | cluster_service:2278 - _execute_script_in_nodes | Request Id: fc8e70a8-38ae-4a2f-8649-3bf2edf0d5b0 | DEBUG :: error: unable to drain node "node-mpgt", aborting command.                                                                 ..

There are pending nodes to be drained:
 node-mpgt
error: cannot delete Pods not managed by ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet (use --force to override): olm/operatorhubio-catalog-f9rcs

21-07-07 09:58:44 | cluster_service:2279 - _execute_script_in_nodes | Request Id: fc8e70a8-38ae-4a2f-8649-3bf2edf0d5b0 | DEBUG :: node/node-mpgt cordoned

21-07-07 09:58:44 | cluster_service:1790 - _drain_nodes | Request Id: fc8e70a8-38ae-4a2f-8649-3bf2edf0d5b0 | ERROR :: Failed to drain nodes ['node-mpgt'] in cluster 'test-c24' (vapp: https                                                                 ://iaas-vdc.hi.inet/api/vApp/vapp-741c469c-3c37-4273-bba6-e224664a38d1) with error: Script execution failed on node ['mstr-1dd9']
Errors: ['error: unable to drain node "node-mpgt", aborting command...\n\nThere are pending nodes to be drained:\n node-mpgt\nerror: cannot delete Pods not managed by ReplicationController                                                                 , ReplicaSet, Job, DaemonSet or StatefulSet (use --force to override): olm/operatorhubio-catalog-f9rcs\n']
21-07-07 09:58:44 | cluster_service:1397 - _upgrade_cluster_async | Request Id: fc8e70a8-38ae-4a2f-8649-3bf2edf0d5b0 | ERROR :: Unexpected error while upgrading cluster 'test-c24'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/container_service_extension/def_/cluster_service.py", line 1302, in _upgrade_cluster_async
    cluster_name=cluster_name)
  File "/usr/local/lib/python3.7/site-packages/container_service_extension/def_/cluster_service.py", line 1788, in _drain_nodes
    script)
  File "/usr/local/lib/python3.7/site-packages/container_service_extension/def_/cluster_service.py", line 2310, in _run_script_in_nodes
    raise E.ScriptExecutionError(f"Script execution failed on node "
container_service_extension.exceptions.ScriptExecutionError: Script execution failed on node ['mstr-1dd9']
Errors: ['error: unable to drain node "node-mpgt", aborting command...\n\nThere are pending nodes to be drained:\n node-mpgt\nerror: cannot delete Pods not managed by ReplicationController                                                                 , ReplicaSet, Job, DaemonSet or StatefulSet (use --force to override): olm/operatorhubio-catalog-f9rcs\n']
...
```

As per the issues:
https://github.com/operator-framework/operator-lifecycle-manager/issues/1514
https://github.com/kubernetes/kubernetes/issues/57049

To overcome this error, a `kubectl drain ... --force` couldbe run, but since the upgrade process drains the nodes (master & workers) several times during the upgrade, it is not possible to _intercept_ all of the times that it would be required.
 
If a `--force` is added to the following 
 - 'vcdbroker.py'-> line 1402
 - 'cluster_service.py'-> line 1780
 ```python
        script += f"kubectl drain {node_name} " \
                  f"--ignore-daemonsets --timeout=60s --delete-local-data --force\n"
```

The upgrade does suceed for these types of clusters.
Given the popularity of OLM, this would be quite desirable :)

**Comment by goelaashima** [comment]
Hi Dario

We are investigating it on our side. Thanks for reporting it. 

Aashima

**Comment by Anirudh9794** [comment]
Hello @dmbuil ,
The issue should be resolved in CSE 3.0.4.
I am closing this issue for now but if you still see the issue, feel free to re-open.

Thank you,
Aniruddha
